SMTPServer/MailSession.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+#include <iostream>
+
+#include <winsock2.h>
+#include <ws2tcpip.h>
+#include <string>
+
+#include "MailInfo.h"
+
+#pragma comment(lib, "Ws2_32.lib")
+
+#define MAX_ADDRESS_LENGTH 256
+#define SMTP_DATA_TERMINATOR "\r\n.\r\n"
+#define FIRST_FOUR_SYMBOLS 4
+#define FIRST_EIGHT_SYMBOLS 8
+
+//#define SMTP_DATA_TERMINATOR "."
+
+enum MailSessionStatus
+{
+	EMPTY,
+	EHLO,
+	MAIL_FROM,
+	RCPT_TO,
+	DATA,
+	SUBJECT,
+	QUIT
+};
+
+enum Responses
```

Comment:
Consider to use enum class instead of plain enum.

Reply:
I don't need to change it, bacause I often cast my enum types to int and I compare to different enums.

Reply:
You can cast enum class types to int as well.